etcdmain/etcd.go

```diff
@@ -146,6 +147,19 @@ func Main() {
 			plog.Errorf("please check (cURL) the discovery token for more information.")
 			plog.Errorf("please do not reuse the discovery token and generate a new one to bootstrap the cluster.")
 		default:
+			if strings.Contains(err.Error(), "match") && strings.Contains(err.Error(), "--initial-cluster") {
+				plog.Infof("%v", err)
+				if cfg.initialCluster == initialClusterFromName(cfg.name) {
```

Comment:
No, I would not expect so. I think leaving a TODO is more elegant, and today's one is good.